yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/reader/Tokenizer.scala

```diff
@@ -109,9 +108,22 @@ private[yaml] class Scanner(str: String) extends Tokenizer {
     val name = readAnchorName()
     (name, pos)
 
-  private def parseAnchor() =
-    val (name, pos) = parseAnchorName()
-    Token(Anchor(name), pos)
+  private def parseAnchor(): List[Token] =
+    val (name, anchorPos) = parseAnchorName()
+    val nexTokens         = getNextTokens()
+
+    val anchorToken = Token(Anchor(name), anchorPos)
+    nexTokens match {
+      case (Token(_: MappingStart.type, _) |
+          Token(_: FlowMappingStart.type, _)) :: Token(_: MappingKey.type, _) :: rest =>
+        ctx.removeLastIndent()
+        ctx.addIndent(anchorPos.column)
+        nexTokens.take(2) ::: anchorToken +: rest
+      case Token(_: MappingKey.type, _) :: rest
+          if (ctx.indent == anchorPos.column || ctx.isInFlowCollection) =>
+        nexTokens.take(1) ::: anchorToken +: rest
```

Comment:
Which situation is handled by this case?

Reply:
```
key: value
&a a: &b b
```